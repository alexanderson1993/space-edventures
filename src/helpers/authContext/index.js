import React from "react";

const AuthContext = React.createContext({
  user: null,
  login: () => {},
  logout: () => {}
});

export default AuthContext;
