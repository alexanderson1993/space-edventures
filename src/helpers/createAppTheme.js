import { lighten, darken } from "polished";

const generateColor = color => ({
  base: color,
  light: lighten(0.2, color),
  dark: darken(0.2, color)
});

const generateBackground = color => ({
  level0: color,
  level1: lighten(0.015, color),
  level2: lighten(0.03, color),
  level3: lighten(0.045, color)
});

export default (theme = {}) => ({
  ...theme,
  color: {
    primary: generateColor(theme.colorPrimary || "#30fffe"),
    header: generateColor(theme.colorHeader || "#a1ecfb"),
    control: generateColor(theme.colorControl || "#acf9fb"),
    ...theme.color
  },
  background: {
    primary: generateBackground(theme.backgroundPrimary || "#031212"),
    ...theme.background
  }
});
