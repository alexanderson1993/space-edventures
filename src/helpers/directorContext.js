import React from "react";
export const DirectorContext = React.createContext({});
