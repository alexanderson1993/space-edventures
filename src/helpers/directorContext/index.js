import React, { useContext, useMemo } from "react";
import propTypes from "prop-types";
import AuthContext from "../authContext";
import { Query } from "react-apollo";
import graphQLHelper from "../graphQLHelper";
import CENTER_DIRECTOR from "./centerDirector.graphql";
import { Match } from "@reach/router";
export const DirectorContext = React.createContext({});

const Provider = ({ me = {}, children }) => {
  const value = useMemo(() => me, [me]);
  return (
    <DirectorContext.Provider value={value}>
      {children}
    </DirectorContext.Provider>
  );
};
export const DirectorProvider = ({ children }) => {
  const { user } = useContext(AuthContext);
  return (
    <Match path="/director/:centerId/*">
      {({ match }) => (
        <Query
          query={CENTER_DIRECTOR}
          variables={{
            centerId: match && match.centerId,
            id: user && user.id
          }}
          skip={!user || !user.id}
        >
          {graphQLHelper(({ me }) => (
            <Provider me={me} children={children} />
          ))}
        </Query>
      )}
    </Match>
  );
};

DirectorProvider.propTypes = {
  children: propTypes.node
};
