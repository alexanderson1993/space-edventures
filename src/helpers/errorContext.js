import React, { useReducer, useMemo } from "react";
export const ErrorContext = React.createContext();

const initialState = { error: false };
function reducer(state, action) {
  switch (action.type) {
    case "errored":
      return {
        error: true,
        errorData: action.errorData,
        errorInfo: action.errorInfo
      };
    case "cleared":
      return { error: false, errorData: null, errorInfo: null };
    default:
      throw new Error();
  }
}

export const ErrorProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);
  const value = useMemo(() => ({ state, dispatch }), [state]);
  return (
    <ErrorContext.Provider value={value}>{children}</ErrorContext.Provider>
  );
};
