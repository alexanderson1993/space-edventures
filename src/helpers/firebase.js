import { firebase } from "@firebase/app";
import "@firebase/auth";

var config = {
  apiKey: process.env.GATSBY_FIREBASE_API_KEY,
  databaseURL: process.env.GATSBY_FIREBASE_DATABASE_URL,
  storageBucket: process.env.GATSBY_FIREBASE_STORAGE_BUCKET,
  authDomain: process.env.GATSBY_FIREBASE_AUTH_DOMAIN,
  messagingSenderId: process.env.GATSBY_FIREBASE_MESSAGE_SENDER_ID,
  projectId: process.env.GATSBY_FIREBASE_PROJECT_ID
};
let authItem;
let baseAuthItem;
if (typeof window !== "undefined") {
  firebase.initializeApp(config);
  authItem = firebase.auth();
  baseAuthItem = firebase.auth;
}
export const auth = authItem;
export const baseAuth = baseAuthItem;
