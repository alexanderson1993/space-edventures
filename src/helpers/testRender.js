import React from "react";
import { render as rtlRender } from "react-testing-library";
import { ThemeProvider, createTheme } from "@arwes/arwes";
import { SoundsProvider, createSounds } from "@arwes/sounds";
import createAppTheme from "./createAppTheme";
import { AdminContext } from "../layout/arwesProvider";
import AuthContext from "./authContext/index";

const normalTheme = createTheme(
  createAppTheme({
    typography: {
      headerSizes: {
        h1: 44
      }
    }
  })
);

const ArwesProvider = ({ children }) => {
  return (
    <AuthContext.Provider
      value={{
        completeMagicLinkSignin: () => Promise.resolve()
      }}
    >
      <SoundsProvider sounds={createSounds({})}>
        <ThemeProvider theme={normalTheme}>
          <AdminContext.Provider value={{ isAdmin: false }}>
            {children}
          </AdminContext.Provider>
        </ThemeProvider>
      </SoundsProvider>
    </AuthContext.Provider>
  );
};

export function render(ui) {
  return rtlRender(<ArwesProvider>{ui}</ArwesProvider>);
}
