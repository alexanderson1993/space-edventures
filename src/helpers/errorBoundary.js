import React, { Component } from "react";
import { Button } from "../components";
import styled from "@emotion/styled";
import { ErrorContext } from "./errorContext";

const Container = styled("div")`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;
export default class ErrorBoundary extends Component {
  static contextType = ErrorContext;
  componentDidCatch(error, errorInfo) {
    // Catch errors in any components below and re-render with error message
    this.context.dispatch({ type: "errored", errorData: error, errorInfo });
  }
  refresh() {
    if (typeof window !== "undefined") {
      localStorage.clear();
      window.location.reload();
    }
  }
  render() {
    if (this.context.state.error) {
      // Error path
      if (this.props.render) return this.props.render;
      return (
        <div className="error-boundary">
          <Container>
            <h2>Error in the Space EdVentures app.</h2>
            <Button onClick={this.refresh}>Refresh</Button>

            <h3>
              If you know what you are looking for, you can check out what went
              wrong.
            </h3>
            <details style={{ whiteSpace: "pre-wrap" }}>
              {this.context.state.errorData &&
                this.context.state.errorData.toString()}
              <br />
              {this.context.state.errorInfo.componentStack}
            </details>
          </Container>
        </div>
      );
    }
    // Normally, just render children
    return this.props.children;
  }
}
