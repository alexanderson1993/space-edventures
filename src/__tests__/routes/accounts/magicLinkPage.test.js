import React from "react";
import { render } from "../../../helpers/testRender";
import MagicLinkPage from "../../../pages/accounts/magicLink";
import { ErrorProvider } from "../../../helpers/errorContext";

test("magicLinkPage", () => {
  const { container } = render(
    <ErrorProvider>
      <MagicLinkPage location={{ href: "/" }} />
    </ErrorProvider>
  );
  expect(container).toMatchSnapshot();
});
