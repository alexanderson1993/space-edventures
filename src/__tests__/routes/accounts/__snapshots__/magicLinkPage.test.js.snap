// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`magicLinkPage 1`] = `
.emotion-0 {
  margin: 0 auto;
  width: 960px;
  max-width: 80%;
  padding: 0px 1.0875rem 1.45rem;
  padding-top: 1.45rem;
  margin-bottom: 10%;
}

@media only screen and (max-width:600px) {
  .emotion-0 {
    padding: 0px 0.5875rem 1.45rem;
    max-width: 98%;
  }
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <h1>
      <span
        class="Sounds(Words)-root-0-1-1 Sounds(Words)-root-0-1-7 Sounds(Words)-hide-0-1-5"
      >
        <span
          class="Sounds(Words)-children-0-1-2"
        >
          Logging In...
        </span>
      </span>
    </h1>
    <div
      class="Loading-root-0-1-8 Loading-entering-0-1-14"
    >
      <div
        class="Loading-circle-0-1-9 Loading-circle-0-1-18 Loading-circle1-0-1-10"
      />
      <div
        class="Loading-circle-0-1-9 Loading-circle-0-1-18 Loading-circle2-0-1-11"
      />
    </div>
  </div>
</div>
`;
